File: Week-5/MyApplication/app/src/main/java/com/example/myapplication/MainActivity2.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

public class MainActivity2 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        String message = "Xin chào từ MainActivity";
        Intent intent = new Intent(this, HelloIntentService.class);
        intent.putExtra("message", message);
        startService(intent);

        Button startServiceButton = findViewById(R.id.start_service_button);
        Button stopServiceButton = findViewById(R.id.stop_service_button);

        startServiceButton.setOnClickListener(view -> startHelloService());

        stopServiceButton.setOnClickListener(view -> stopHelloService());


    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case 2131230949:
                Intent intent = new Intent(this, MainActivity.class);
                startActivity(intent);
                break;
            case 2131230950:
                break;
        }
        return true;
    }

    private void startHelloService() {
        Intent serviceIntent = new Intent(this, HelloService.class);
        startService(serviceIntent);
    }

    private void stopHelloService() {
        Intent serviceIntent = new Intent(this, HelloService.class);
        stopService(serviceIntent);
    }
}

File: Week-3/Practice/app/src/main/java/com/example/myapplication/MainActivity2.kt
package com.example.practice

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.example.myapplication.R

class MainActivity2 : AppCompatActivity() {
    private var send_message_button: Button? = null
    private var edit_text: EditText? = null
    private var MY_REQUEST_CODE = 10
    private var text_feedback: TextView? = null
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        this.edit_text = findViewById<EditText>(R.id.edit_text)
        this.send_message_button = findViewById<Button>(R.id.send_button);
        this.text_feedback = findViewById<TextView>(R.id.text_feedback)
        send_message_button?.setOnClickListener {
            sendMessage()
        }

        val backButton = findViewById<Button>(R.id.go_back)
        backButton.setOnClickListener {
            this.onBackPressed();
        }
    }
    private fun sendMessage() {
        val fullName = edit_text?.text.toString();
        val message = "Hello, Please say hello me!";

        val intent = Intent(this, MainActivity3::class.java)
        intent.putExtra("fullName", fullName)
        intent.putExtra("message", message)
        this.startActivityForResult(intent, MY_REQUEST_CODE)
    }

    override fun onActivityResult(requestCode : Int, resultCode : Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (resultCode == Activity.RESULT_OK && requestCode == MY_REQUEST_CODE) {
            val feedback = data?.getStringExtra("feedback")
            this.text_feedback?.text = feedback
        } else {
            this.text_feedback?.text = "!?"
        }
    }

}